src/Features/Core/Portable/ConvertToInterpolatedString/AbstractConvertRegularStringToInterpolatedStringRefactoringProvider.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.Editing;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.ConvertToInterpolatedString
+{
+    /// <summary>
+    /// Code refactoring that converts a regular string containing braces to an interpolated string
+    /// </summary>
+    internal abstract class AbstractConvertRegularStringToInterpolatedStringRefactoringProvider<TExpressionSyntax> : CodeRefactoringProvider
+        where TExpressionSyntax : SyntaxNode
+    {
+        public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
+        {
+            var literalExpression = await context.TryGetRelevantNodeAsync<TExpressionSyntax>().ConfigureAwait(false);
+            if (literalExpression == null || !IsAppropriateLiteralKind(literalExpression))
```

Comment:
instead of this, take the positoin, and get hte token underneath it.  Then ensure your position is within the bounds (not full-bounds) of that token.

Then, you can use ISyntaxKindsService to check if this is a string literal.

Reply:
Is this what you had in mind?

```cs
public override async Task ComputeRefactoringsAsync(CodeRefactoringContext context)
{
    var (document, _, cancellationToken) = context;

    var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
    if (root is null)
        return;

    var token = root.FindToken(context.Span.Start);
    if (!context.Span.IntersectsWith(token.Span))
        return;

    var syntaxKinds = document.GetLanguageService<ISyntaxKindsService>();
    if (syntaxKinds is null)
        return;

    if (token.RawKind != syntaxKinds.StringLiteralToken)
        return;

    ...
}
```

Reply:
that's exactly it :)

Reply:
although, you don't need the `syntaxKinds is null` check.  just use `GetRequriedLanguageService.`

Reply:
Ok, done.